File: Resilio Connect API/Python/communication.py
import json
import requests

mcURL = ""
mcPort = -1
mcToken = ""

def initializeMCParams(url, port, token):
  global mcURL
  global mcPort
  global mcToken
  mcURL = url
  mcPort = port
  mcToken = "Token " + token

def getAPIRequest(APIReq) -> json:
    URL = mcURL + ":" + str(mcPort) + APIReq
    req = requests.get(URL, headers={"Authorization": mcToken})
    return json.loads(req.content)

def postAPIRequest(APIReq, bodyData) -> json:
    URL = mcURL + ":" + str(mcPort) + APIReq
    bodyData = json.dumps(bodyData)
    headersData = {
        "Authorization": mcToken,
        "Content-Type": "application/json",
        "Content-Length": str(len(bodyData))
    }
    req = requests.post(URL, headers=headersData, data=bodyData)
    return json.loads(req.content)

File: Resilio Connect API/JavaScript/communication.js
// @ts-check

module.exports = {
    initializeMCParams,
    getAPIRequest,
    postAPIRequest,
    deleteAPIRequest,
};

var https = require('https');

var mcURL = "";
var mcPort = 0;
var mcToken = "";

/**
 * 
 * @param {string} url 
 * @param {number} port 
 * @param {string} token 
 */
function initializeMCParams(url, port, token) {
  mcURL = url;
  mcPort = port;
  mcToken = "Token " + token;
}

function getAPIRequest(APIReq) {
  var APIResponse = (resolve, reject) => {
    var options = {
      headers: { "Authorization": mcToken },
      host: mcURL,
      port: mcPort,
      path: APIReq
    }

    https.get(options, function (res) {
      var data = '';

      res.on('data', function (chunk) {
        data += chunk;
      });

      res.on('end', () => {
        resolve(data);

      }).on('error', (err) => {
        console.log("there was an error:" + err);
        reject(err);
      });

    });
  }
  return new Promise(APIResponse)
}


function APIRequest(APIReq, method, bodyData) {
  var APIResponse = (resolve, reject) => {
    if (bodyData.length != "") {
      bodyData = JSON.stringify(bodyData);
    }
    var options = {
      headers: {
        "Authorization": mcToken,
        "Content-Type": "application/json",
        "Content-Length": bodyData.length
      },
      host: mcURL,
      port: mcPort,
      path: APIReq,
      method: method,
    }

    const req = https.request(options, function (res) {
      var data = '';

      res.on('data', function (chunk) {
        data += chunk;
      });

      res.on('end', () => {
        resolve(data);

      }).on('error', (err) => {
        console.log("there was an error:" + err);
        reject(err);
      });

    });
    req.write(bodyData);
    req.end();
  }
  return new Promise(APIResponse)
}

function postAPIRequest(APIReq, postData) {
  return APIRequest(APIReq, "POST", postData);
}

function deleteAPIRequest(APIReq) {
  return APIRequest(APIReq, "DELETE", "");
}

